src/main/java/de/hpi/bpt/argos/api/entity/EntityEndpointImpl.java

```diff
@@ -1 +1,349 @@
+		List<Attribute> attributesToInclude = new ArrayList<>();
+
+		for (Attribute entityAttribute : entityAttributes) {
+			if (attributeTypeIds.contains(entityAttribute.getTypeAttributeId())) {
+				attributesToInclude.add(entityAttribute);
+			}
+		}
+
+		return attributesToInclude;
+    }
+
+    /**
+     * This method returns a list of entities as a json array including the given attributes.
+     * @param entities the entities to be returned as json
+     * @param attributesTypeIdsToInclude the unique identifiers of the typeAttributes to be returned in json
+     * @return json array that represents the given entities
+     */
+    private JsonArray getEntitiesJson(List<Entity> entities, List<Long> attributesTypeIdsToInclude) {
+        JsonArray entitiesJson = new JsonArray();
+        for (Entity entity : entities) {
+            entitiesJson.add(getEntityJson(entity, getIncludedAttributes(entity, attributesTypeIdsToInclude)));
+        }
+        return entitiesJson;
+    }
+
+    /**
+     * This method returns an event type as a JsonObject.
+     * @param eventType - the event type
+     * @return - a json representation of the event type
+     */
+    private JsonObject getEventTypeJson(EventType eventType) {
+        try {
+            JsonObject jsonEventType = new JsonObject();
+
+            jsonEventType.addProperty("Id", eventType.getId());
+            jsonEventType.addProperty("Name", eventType.getName());
+            jsonEventType.addProperty("NumberOfEvents",
+                    PersistenceAdapterImpl.getInstance().getEventCountOfEventType(eventType.getId()));
+            jsonEventType.addProperty("TimestampAttributeId", eventType.getTimeStampAttributeId());
+
+            return jsonEventType;
+        } catch (Exception e) {
+            LoggerUtilImpl.getInstance().error(logger, "cannot parse event type", e);
+            return new JsonObject();
+        }
+    }
+
+    /**
+     * This method returns event types as a JsonArray.
+     * @param eventTypes - the event types
+     * @return - a json representation of the event types
+     */
+    private JsonArray getEventTypesJson(List<EventType> eventTypes) {
+        JsonArray eventTypesJson = new JsonArray();
+        for (EventType eventType : eventTypes) {
+            eventTypesJson.add(getEventTypeJson(eventType));
+        }
+        return eventTypesJson;
+    }
+
+    /**
+     * This method returns events as a JsonArray.
+     * @param events - the events
+     * @return - a json representation of the events
+     */
+    private JsonArray getEventsJson(List<Event> events) {
+        JsonArray eventsJson = new JsonArray();
+        for (Event event : events) {
+            List<Attribute> attributes = PersistenceAdapterImpl.getInstance().getAttributes(event.getId());
+            JsonArray jsonAttributes = getAttributesJson(attributes);
+
+            JsonObject eventJson = new JsonObject();
+            eventJson.add("Attributes", jsonAttributes);
+
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER') NullPointerException might be thrown as 'entityNode' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS2259)